src/App.tsx

```diff
@@ -17,18 +73,35 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onSelectChange={handleSelectChange}
+                query={query}
+                onInputChange={handleInputChange}
+                resetInput={resetInput}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+
+              {!loading && (
+                <TodoList
+                  todos={visibleTodos}
+                  onSelectTodo={handleSelectTodo}
+                  selectedTodo={selectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          onResetSelectedTodo={handleResetSelectedTodo}
```

Comment:
The key 'onResetSelectedTodo' does not follow the naming conventions for event handlers in React. Please refer to the link provided in the checklist for the recommended naming conventions.